wandb/run_manager.py

```diff
@@ -430,22 +430,52 @@ def format_run_name(run):
     "Simple helper to not show display name if its the same as id"
     return " "+run.name+":" if run.name != run.id else ":"
 
+
+class RunStatusChecker(object):
+    """Polls the backend periodically to check on this run's status.
+
+    For now, we just use this to figure out if the user has requested a stop.
+    TODO(adrnswanberg): Use this as more of a general heartbeat check.
+    """
+    def __init__(self, run, api, stop_requested_handler, polling_interval=15):
+        self._run = run
+        self._api = api
+        self._polling_interval = polling_interval
+        self._stop_requested_handler = stop_requested_handler
+
+        self._thread = threading.Thread(target=self.check_status)
+        self._thread.daemon = True
+        self._thread.start()
+
+    def check_status(self):
+        while True:
+            should_exit =  self._api.check_stop_requested(
+                project_name=self._run.project_name(),
```

Comment:
I *believe @normalize_exceptions will return a CommError if the request fails.  We should catch whatever errors we need to here to handle 5xx’s or network issues.

Reply:
You mean just catch the exception and print something? If so, done.